### DISCUSSION TITLE
Pyright convert typing.Union to new PEP 604 syntax

### DISCUSSION THREAD

**Discussion by kumaraditya303** [microsoft/pyright#2550] [Ideas]
Feature request to convert legacy typing.Union syntax to newer PEP 604 syntax for large codebases where manually converting will take a lot of effort. Thanks

**Comment by erictraut** [comment]
This isn't a core type checking feature, so it's not something we'd look to add to pyright. If you would like to see this feature in pylance, I recommend filing a feature request in the [pylance-release issue tracker](https://github.com/microsoft/pylance-release/issues).

**Reply by kumaraditya303** [reply]
I am interested more in cli based system so that it can be done easily, in pylance however it would requires vscode.

**Reply by kumaraditya303** [reply]
It would be a great feature and a selling point for pyright as more codebases would need to be migrated as typing.Union is going to be deprecated in later versions of Python AFAIK 3.10.

**Reply by erictraut** [reply]
`Union` isn't going away any time soon. There's no need to worry about that.

This functionality falls outside of the bounds of pyright's intended usage, so we're unlikely to implement a feature like this. It's more closely aligned to the functionality provided by pylance, which provides programming efficiency features including various code actions.

**Reply by kumaraditya303** [reply]
Created https://github.com/microsoft/pylance-release/issues/2055

**Reply by hmc-cs-mdrissi** [reply]
pyupgrade offers this https://github.com/asottile/pyupgrade#pep-604-typing-rewrites as a simple cli and is usable as a pre-commit hook as an easy way to keep files up dated. pyupgrade does other transformations though and unsure you can pick which ones.
